pkg/config/metadata.go

```diff
@@ -0,0 +1,91 @@
+package config
+
+import (
+	"embed"
+	"fmt"
+	"io/fs"
+	"regexp"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/k0sproject/k0s/pkg/constant"
+	"github.com/replicatedhq/embedded-cluster/pkg/release"
+	"github.com/replicatedhq/embedded-cluster/pkg/versions"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	_metadata   *release.K0sMetadata
+	metadataMap = map[string]release.K0sMetadata{}
+
+	//go:embed static/metadata-1_*.yaml
+	metadataFS embed.FS
+
+	metadataMinorRegex = regexp.MustCompile(`^metadata-1_(\d+)\.yaml$`)
+)
+
+// Metadata returns the metadata for the given k0s minor version.
+func Metadata(minorVersion string) release.K0sMetadata {
+	metadata, ok := metadataMap[minorVersion]
+	if !ok {
+		panic(fmt.Sprintf("no metadata found for k0s version: %s", minorVersion))
+	}
+	return metadata
+}
+
+func init() {
+	if err := populateMetadataMap(); err != nil {
+		panic(fmt.Errorf("failed to populate metadata map: %v", err))
+	}
+
+	k8sVersion, err := semver.NewVersion(constant.KubernetesMajorMinorVersion)
```

Comment:
where is this constant coming from?

Reply:
this makes sure that the k0s go dep is compatible with the tag version identifier from LDFLAGS